src/dex/nerve/types.ts

```diff
@@ -0,0 +1,81 @@
+import { JsonFragment } from '@ethersproject/abi';
+import type { DeepReadonly } from 'ts-essentials';
+import type { BlockHeader } from 'web3-eth';
+import type { Address, Log, Token } from '../../types';
+import type { NerveEventMetapool } from './nerve-metapool';
+import type { NerveEventPool } from './nerve-pool';
+
+export interface PoolState {
+  initialA: bigint;
+  futureA: bigint;
+  initialATime: bigint;
+  futureATime: bigint;
+  swapFee: bigint;
+  adminFee: bigint;
+  defaultDepositFee?: bigint;
+  defaultWithdrawFee?: bigint;
+  lpToken_supply: bigint;
+  balances: bigint[];
+  tokenPrecisionMultipliers: bigint[];
+  isValid: boolean;
+}
```

Comment:
```suggestion
export type PoolState = {
  initialA: bigint;
  futureA: bigint;
  initialATime: bigint;
  futureATime: bigint;
  swapFee: bigint;
  adminFee: bigint;
  defaultDepositFee?: bigint;
  defaultWithdrawFee?: bigint;
  lpToken_supply: bigint;
  balances: bigint[];
  tokenPrecisionMultipliers: bigint[];
  isValid: true;
} | { isValid: false }
```
This can be used to avoid cloning the state

Reply:
Fixed another way as we discussed